src/view/uniprotkb/components/protvista-config/variationFilters.ts

```diff
@@ -0,0 +1,215 @@
+      getFilteredVariants(variants, (variantPos: TransformedProtvistaVariant) =>
+        significanceMatches(
+          consequences.likelyBenign,
+          variantPos.clinicalSignificances
+        )
+      ),
+  },
+  {
+    name: 'uncertain',
+    type: {
+      name: 'consequence',
+      text: 'Filter Consequence',
+    },
+    options: {
+      labels: ['Uncertain'],
+      colors: [scaleColors.othersColor],
+    },
+    filterData: (variants: VariationVariants) =>
+      getFilteredVariants(
+        variants,
+        (variantPos: TransformedProtvistaVariant) =>
+          (typeof variantPos.clinicalSignificances === 'undefined' &&
+            typeof variantPos.polyphenScore === 'undefined' &&
+            typeof variantPos.siftScore === 'undefined') ||
+          significanceMatches(
+            consequences.uncertain,
+            variantPos.clinicalSignificances
+          )
+      ),
+  },
+  {
+    name: 'UniProt',
+    type: {
+      name: 'provenance',
+      text: 'Filter Provenance',
+    },
+    options: {
+      labels: ['UniProt reviewed'],
+      colors: ['#9f9f9f'],
+    },
+    filterData: (variants: VariationVariants) =>
+      getFilteredVariants(
+        variants,
+        (variantPos: TransformedProtvistaVariant) =>
+          variantPos.xrefNames &&
+          (variantPos.xrefNames.includes('uniprot') ||
+            variantPos.xrefNames.includes('UniProt'))
+      ),
+  },
+  {
+    name: 'ClinVar',
+    type: {
+      name: 'provenance',
+      text: 'Filter Provenance',
+    },
+    options: {
+      labels: ['ClinVar reviewed'],
+      colors: ['#9f9f9f'],
+    },
+    filterData: (variants: VariationVariants) =>
+      getFilteredVariants(
+        variants,
+        (variantPos: TransformedProtvistaVariant) =>
+          variantPos.xrefNames &&
+          (variantPos.xrefNames.includes('ClinVar') ||
+            variantPos.xrefNames.includes('clinvar'))
+      ),
+  },
+  {
+    name: 'LSS',
+    type: {
+      name: 'provenance',
+      text: 'Filter Provenance',
+    },
+    options: {
+      labels: ['Large scale studies'],
+      colors: ['#9f9f9f'],
+    },
+    filterData: (variants: VariationVariants) =>
+      getFilteredVariants(
+        variants,
+        (variantPos: TransformedProtvistaVariant) =>
+          variantPos.sourceType === 'large_scale_study' ||
+          variantPos.sourceType === 'mixed'
+      ),
+  },
+];
+
+const predictionScale = scaleLinear<string>()
+  .domain([0, 1])
+  .range([scaleColors.deleteriousColor, scaleColors.benignColor]);
+
+const getPredictionColor = (siftScore?: number, polyphenScore?: number) => {
+  return predictionScale(
+    (siftScore || 0 + (1 - (polyphenScore || 1))) /
+      (polyphenScore && siftScore ? 2 : 1)
+  );
+};
+
+export const colorConfig = (variant: TransformedProtvistaVariant) => {
+  const variantWrapper = [{ variants: [variant] }];
+  const filterGroups = groupBy(filterConfig, (config) => config.name);
```

Comment:
What issue are we trying to address? Is it because of accessing the zeroth element in the `if` statements? Or the fact the `groupBy` has to run for every variant?

The issue with 1. is that name would then be both the key but also one of the attributes (this is required by the protvista-filter component).

Reply:
1. **Readability** There are several times where the zeroth element is accessed in a single line: `filterGroups.disease[0].filterData(variantWrapper)[0].variants.length > 0`. I had to look into this a bit because I was thinking could `filterGroups.disease` be an empty array (`!!(new Array()) = true`) and so trying to access the 0th element would raise an exception? Now I understand that it will work because I looked at the `filterConfig` array of objects but it wasn't immediately obvious.
2. **Efficiency** As loading the entry P05067 calls `groupBy` ~1K times and changing the sequence range had ~7K calls I think it's worthwhile trying optimise the `colorConfig` function. If it's possible to avoid using the `groupBy` I think that will help do this.

> The issue with 1. is that name would then be both the key but also one of the attributes (this is required by the protvista-filter component).

If we use `Object.values(filterConfig)` then it won't have the names as keys as it would just be an array, like what is currently exported from `variationFilters.ts`.

Reply:
Makes sense, changes made.